src/core/libmaven/csvreports.cpp

```diff
@@ -1,283 +1,155 @@
-
-}
-
-void CSVReports::initialCheck(string outputfile) {
-
-    if (samples.size() == 0)
-        return;
-    if (QString(outputfile.c_str()).endsWith(".csv", Qt::CaseInsensitive))
-        setCommaDelimited();
+    sort(samples.begin(), samples.end(), mzSample::compSampleOrder);
 }
 
-void CSVReports::openGroupReportCSVFile(string outputfile) {
+CSVReports::~CSVReports() {
 
-     groupReport.open(outputfile.c_str());
 }
 
-void CSVReports::openPeakReportCSVFile(string outputfile) {
+bool CSVReports::exportGroup(){
 
-     peakReport.open(outputfile.c_str());
-}
-
-void CSVReports::insertGroupReportColumnNamesintoCSVFile(string outputfile,bool includeSetNamesLine){
-    if (groupReport.is_open()) {
-        QStringList groupReportcolnames;
-        groupReportcolnames << "label" << "metaGroupId" << "groupId" << "goodPeakCount"
-                << "medMz" << "medRt" << "maxQuality" << "isotopeLabel" << "compound"
-                << "compoundId" << "formula" << "expectedRtDiff" << "ppmDiff" << "parent";
-        int cohort_offset = groupReportcolnames.size() - 1;
-        QString header = groupReportcolnames.join(SEP.c_str());
-        groupReport << header.toStdString();
-        for (unsigned int i = 0; i < samples.size(); i++) {
-            groupReport << SEP << sanitizeString(samples[i]->sampleName.c_str()).toStdString();
-        }
-        groupReport << endl;
-        //TODO: Remove this to remove row in csv reports --@Giridhari
-        if (includeSetNamesLine){
-             for(unsigned int i = 0; i < cohort_offset; i++) { groupReport << SEP; }
-             for(unsigned int i = 0; i < samples.size(); i++) { groupReport << SEP << sanitizeString(samples[i]->getSetName().c_str()).toStdString(); }
-             groupReport << endl;
-         }
+    if(groups.size() == 0){
+        errorReport = "No groups found!";
+        return 0;
     }
-    else {
-        errorReport = "Unable to write to file \"" + QString::fromStdString(outputfile) + "\"\n";
-        errorReport += "Please check if you have permission to write to the specified location or the file is not in use";
+    if(_fileName.length() < 4){
+        errorReport="invalid file name";
+        return 0;
     }
-}
-
-void CSVReports::insertPeakReportColumnNamesintoCSVFile(){
-
-    if (peakReport.is_open()) {
-        QStringList peakReportcolnames;
-        peakReportcolnames << "groupId" << "compound" << "compoundId" << "formula" << "sample" << "peakMz"
-                << "medianMz" << "baseMz" << "rt" << "rtmin" << "rtmax" << "quality"
-                << "peakIntensity" << "peakArea" << "peakSplineArea" << "peakAreaTop"
-                << "peakAreaCorrected" << "peakAreaTopCorrected"
-                << "noNoiseObs" << "signalBaseLineRatio"
-                << "fromBlankSample";
-        QString header = peakReportcolnames.join(SEP.c_str());
-        peakReport << header.toStdString() << endl;
+    string fileFormat = _fileName.substr(_fileName.length()-4 , 4 );
```

Comment:
@IpankajI Just a suggestion, can we set a delimiter type when user chooses which format they want and just use it everywhere we want? It would save on all the string matching etc.